### PR TITLE
fix: Allow AuthType None to use valid API tokens

### LINKED ISSUE #5799: When AuthType is `none`, `api/client/features` ignores tokens
### Describe the bug

If you set `AuthType.None`, the `api/client/features` endpoint will ignore any auth token you pass to it. This results in it always returning data for all projects and the `default` environment. 

In other words, it is **not possible** to get features for a different environment or to get a payload that only includes a subset of all the projects that exist.

Somewhat related to #5785.

### Steps to reproduce the bug

1. Start Unleash with AuthType=None (either through env variables or through the Unleash auth options)
2. Have a project with some flags in it. Define configuration for at least one of the flags in the `development` environment (or any non-`default` environment) and enable it for that environment. Make sure the flag is not enabled for the `default` env (shouldn't be possible unless you reactivate the env)
3. Create a token for that environment (or re-use an existing one)
4. Using the token you created, request flags from the `api/client/features` endpoint.
5. Inspect the payload, notice that:
  i. The flag has: `enabled: false`
  ii. The flag has an empty list of strategies. No configuration.

### Expected behavior

When you provide an auth token to a client features request, the auth token's env and project settings should be respected.

### Logs, error output, etc.

_No response_

### Screenshots

_No response_

### Additional context

Appears to be because we don't use 

```js
    app.use(baseUriPath, apiTokenMiddleware(config, services));
```

when you use auth type none, so we don't pass the token's data to the client request handler. Adding that line to the handler would fix this one specific issue, but it breaks a lot of tests because it disables the option to call the endpoint with no auth at all. This was attempted in #5764.

The fact that you can call this endpoint without the auth token may not be intended, but it *is* how it works today, so removing it could be considered breaking (especially when we have about 40 tests that rely on that behavior).

### Unleash version

Tip of main a couple days ago.

### Subscription type

None

### Hosting type

None

### SDK information (language and version)

_No response_

### PR DESCRIPTION
**Pull request by ivarconr**
Fixes #5799 and #5785

When you do not provide a token we should resolve to the "default" environment to maintain backward compatibility. If you actually provide a token we should prefer that and even block the request if it is not valid. 

An interesting fact is that "default" environment is not available on a fresh installation of Unleash. This means that you need to provide a token to actually get access to toggle configurations. 

**Todo:**

- [x] Add a few tests to showcase how it is intended to work. 
- [x] Add deprecation warnings in logs and documentation.

### CODE REVIEW

File: src/lib/middleware/no-authentication.ts [hunk 1]
```diff
@@ -12,4 +14,38 @@ function noneAuthentication(basePath: string, app: Application): void {
         next();
     });
 }
+
+export function noApiToken(baseUriPath: string, app: Application) {
+    app.use(`${baseUriPath}/api/frontend`, (req, res, next) => {
+        console.log(req.path);
```
Comment (ivarconr) [inline src/lib/middleware/no-authentication.ts:20]: remove

File: src/lib/openapi/spec/user-schema.ts [hunk 2]
```diff
@@ -11,7 +11,6 @@ export const userSchema = {
         id: {
             description: 'The user id',
             type: 'integer',
-            minimum: 0,
```
Comment (ivarconr) [inline src/lib/openapi/spec/user-schema.ts:14]: This breaks other assumptions.

Comment (daveleek) [reply src/lib/openapi/spec/user-schema.ts:14]: Good point!

**Review by daveleek — APPROVED** [review]
LGTM!

File: src/lib/app.ts [hunk 3]
```diff
@@ -147,6 +147,8 @@ export default async function getApp(
             break;
         }
         case IAuthType.NONE: {
+            noApiToken(baseUriPath, app);
+            app.use(baseUriPath, apiTokenMiddleware(config, services));
```
[the remark this answers is not in the document]
Comment (ivarconr) [reply src/lib/app.ts:151]: I agree that app.ts is complex, but I do not feel this change is making it worse as it follow the same pattern as already established.

File: website/docs/using-unleash/deploy/configuring-unleash.md [hunk 4]
```diff
@@ -63,8 +63,8 @@ unleash.start(unleashOptions);
   - `type` / `AUTH_TYPE`: `string` —  What kind of authentication to use. Possible values
     - `open-source` - Sign in with username and password. This is the default value.
     - `custom` - If implementing your own authentication hook, use this
-    - `none` - Turn off authentication all together
     - `demo` - Only requires an email to sign in (was default in v3)
+    - `none` - _Deprecated_ Turn off authentication all together. If no API token is proved towards /api/client or /api/frontend you will receive configuration for the "default" environment. We generally recommend to use the `demo` type for simple unsecure usage of Unleash. This auth-type has many known limitations, particularly related to personalized capabilities such as "favourites" and "notifications.
```
Comment (kwasniew) [inline website/docs/using-unleash/deploy/configuring-unleash.md:67]: typo: proved -> provided

Comment (kwasniew) [reply website/docs/using-unleash/deploy/configuring-unleash.md:67]: in code we use favorites not favourites so I'd stay consistent

File: src/server-dev.ts [hunk 5]
```diff
@@ -53,6 +53,7 @@ process.nextTick(async () => {
                     },
                 },
                 authentication: {
+                    // type: IAuthType.NONE,
```
Comment (kwasniew) [inline src/server-dev.ts:56]: deliberate?

File: src/test/e2e/api/client/feature.auth-none.e2e.test.ts [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+        {
+            tokenName: 'test',
+            type: ApiTokenType.CLIENT,
+            environment: DEFAULT_ENV,
+            projects: ['default'],
+        },
+    );
+    clientSecret = token.secret;
+
+    const fToken =
```
Comment (kwasniew) [inline src/test/e2e/api/client/feature.auth-none.e2e.test.ts:73]: fToken -> frontendToken

**Review by kwasniew — APPROVED** [review]
Nice!

File: src/server-dev.ts (hunk 5 above)
Comment (ivarconr) [reply src/server-dev.ts:56]: we should not use none developing locally.

File: src/lib/middleware/no-authentication.ts [hunk 7]
```diff
@@ -12,4 +15,37 @@ function noneAuthentication(basePath: string, app: Application): void {
         next();
     });
 }
+
+export function noApiToken(baseUriPath: string, app: Application) {
+    app.use(`${baseUriPath}/api/frontend`, (req, res, next) => {
+        //@ts-ignore
```
[the remark this answers is not in the document]
Comment (ivarconr) [reply src/lib/middleware/no-authentication.ts:21]: yes, requires us to specify the type of the request to potentially have a user.

File: src/test/e2e/api/client/feature.auth-none.e2e.test.ts [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+        .expect(401);
+});
+
+test('returns 3 toggles for correct api token', async () => {
+    return app.request
+        .get('/api/client/features')
+        .set('Authorization', clientSecret)
+        .expect('Content-Type', /json/)
+        .expect(200);
+});
```
Comment (thomasheartman) [inline src/test/e2e/api/client/feature.auth-none.e2e.test.ts:112]: looks like this is missing a check for how many features are returned

File: website/docs/using-unleash/deploy/configuring-unleash.md [hunk 9]
```diff
@@ -63,8 +63,8 @@ unleash.start(unleashOptions);
   - `type` / `AUTH_TYPE`: `string` —  What kind of authentication to use. Possible values
     - `open-source` - Sign in with username and password. This is the default value.
     - `custom` - If implementing your own authentication hook, use this
-    - `none` - Turn off authentication all together
     - `demo` - Only requires an email to sign in (was default in v3)
+    - `none` - _Deprecated_ Turn off authentication all together. If no API token is provided towards /api/client or /api/frontend you will receive configuration for the "default" environment. We generally recommend to use the `demo` type for simple unsecure usage of Unleash. This auth-type has many known limitations, particularly related to personalized capabilities such as "favorites" and "notifications.
```
Comment (thomasheartman) [inline website/docs/using-unleash/deploy/configuring-unleash.md:67]: Good update. Got a few small suggestions:

```suggestion
    - `none` - _Deprecated_ Turn off authentication completely. If no API token is provided towards /`api/client` or `/api/frontend` you will receive configuration for the "default" environment. We generally recommend you use the `demo` type for simple, insecure usage of Unleash. This auth type has many known limitations, particularly related to personalized capabilities such as favorites and [notifications](../../reference/notifications.md). 
```

**Review by thomasheartman — COMMENTED** [review]
This looks good to me, but I'm on board with Mateusz's comments about removing the ts-ignores (or at least trading them for ts-expects with a reason for expecting the error) and the `//authType: 'none'` comment.

File: src/server-dev.ts (hunk 5 above)
Comment (ivarconr) [inline src/server-dev.ts:56]:
```suggestion
```

File: src/test/e2e/api/client/feature.auth-none.e2e.test.ts [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+
+test('returns 401 for incorrect api token', async () => {
+    return app.request
+        .get('/api/client/features')
+        .set('Authorization', 'some-invalid-token')
+        .expect('Content-Type', /json/)
+        .expect(401);
+});
+
+test('returns 3 toggles for correct api token', async () => {
```
Comment (ivarconr) [inline src/test/e2e/api/client/feature.auth-none.e2e.test.ts:106]:
```suggestion
test('returns success for correct api token', async () => {
```

**Pull request merged**